src/nodes.coffee

```diff
@@ -1081,6 +1150,25 @@ exports.Value = class Value extends Base
 
     fragments
 
+  ast: ->
+    ret = @base.ast()
+    for prop, propIndex in @properties
+      ret =
+        mergeAstLocationData(
+          merge {
+            type: 'MemberExpression'
+            object: ret
+            property: prop.ast()
+            computed: prop instanceof Index or prop.name?.unwrap() not instanceof PropertyName
+            optional: !!prop.soak
```

Comment:
So I modeled this `optional` field after what I was seeing in Babel ASTs that used the proposed JS version of the soak operator ("nullish coalescing"). It looks like maybe by the time Babel 7 landed they switched to using different `type`s like `OptionalMemberExpression` rather than just an `optional` field on `MemberExpression`. The `optional` field seems more convenient but it's worth looking into why they chose to model them as unique types

Reply:
See https://github.com/babel/babel/blob/master/packages/babel-parser/ast/spec.md#changes

Reply:
In that link I see a description of the `optional` field but don't see any references to `OptionalMemberExpression`?

Reply:
Oh I don’t know about that specifically, I just noticed it had some spec changes listed.